Scenes/Creeps/Creep.gd

```diff
@@ -497,3 +498,27 @@ func set_spawn_level(spawn_level: int):
 
 func get_special_list() -> Array[int]:
 	return _special_list
+
+
+func set_special_list(special_list: Array[int]):
+	var creep_specials = _creep_specials_container.get_children()
+	creep_specials.clear()
+	for special_id in special_list:
+		var special = WaveSpecial.get_special_buff_map()[special_id]
+		var icon_name = special.get_script().resource_path.get_file().trim_suffix(".gd")
+		var icon_path: String = "res://Scenes/Creeps/Specials/%sSpecial.tscn" % icon_name
+		var icon_texture_rect: PackedScene = load(icon_path) 
+		
+		_creep_specials_container.add_child(icon_texture_rect.instantiate())
+	
+	_special_list = special_list
+
+
+func set_hovered(hovered: bool):
+	super.set_hovered(hovered)
+	_creep_specials_container.set_visible(hovered)
+
+
+func set_selected(selected: bool):
```

Comment:
Current behavior:
1. Hover over creep - special icons visible
2. Select creep - special icons visible
3. Move mouse away while still selecting creep - special icons hidden

Not sure what the intended behavior is. If it's to show special icons while unit is selected, then it doesn't work right now.

Reply:
Not intended. Fixed, so icons should be shown always when creep is selected, regardless of hover.